src/main/java/com/exasol/releasedroid/adapter/communityportal/CommunityPostRenderer.java

```diff
@@ -0,0 +1,95 @@
+package com.exasol.releasedroid.adapter.communityportal;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Renders a Community Post Body.
+ */
+public class CommunityPostRenderer {
+    /**
+     * Render an Exasol release announcement community post body in the format the portal supports.
+     * 
+     * @param projectNameAndVersion name and version of the released project
+     * @param projectDescription    project description
+     * @param changesDescription    description of the changes in the current release
+     * @param gitHubReleaseLink     link to the release on the GitHub
+     * @return rendered community post body
+     */
+    public String renderCommunityPostBody(final String projectNameAndVersion, final String projectDescription,
+            final String changesDescription, final String gitHubReleaseLink) {
+        final var builder = new StringBuilder();
+        builder.append("<h2>About the project</h2>") //
+                .append(formatParagraph(projectDescription)) //
+                .append("<h2>New release</h2>") //
+                .append(formatParagraph(changesDescription)) //
+                .append("<p>For more information check out the <a href=\"") //
+                .append(gitHubReleaseLink) //
+                .append("\" target=\"_blank\" rel=\"noopener\">") //
+                .append(projectNameAndVersion) //
+                .append("</a> release on GitHub.</p>");
+        return builder.toString();
+    }
+
+    private String formatParagraph(final String projectDescription) {
+        final var builder = new StringBuilder();
+        builder.append("<p>");
+        final Queue<Character> chars = createQueue(projectDescription);
+        Character previous = null;
+        while (!chars.isEmpty()) {
+            final char cur = chars.poll();
+            if (cur == '[') {
+                builder.append(renderLink(chars));
```

Comment:
that's dangerous! If `[` appears in text as a nonlink it will bring the converter in an illegal state.

Reply:
If you really want to convert it yourself I would suggest to:
* move it to a separate class (e.g. `MarkdownToHtml`)
* Use Regex where possible (you can just replace markdown links by HTML links)

Reply:
The last one is maybe a matter of taste. Your code is much faster than regex. But I think if you fix all possible issues with [ in the text it will make it totally unreadable. Regex would make it easier, more readable and more robust.